From 26f80ed697256b269b3de167722bf6d668b3bab9 Mon Sep 17 00:00:00 2001
From: kojirou <kojirouhtc@gmail.com>
Date: Mon, 24 May 2021 13:49:34 +0800
Subject: [PATCH] disable file cache

---
 libavutil/file_open.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavutil/file_open.c b/libavutil/file_open.c
index cc302f2f76..a0861aa172 100644
--- a/libavutil/file_open.c
+++ b/libavutil/file_open.c
@@ -86,6 +86,8 @@ int avpriv_open(const char *filename, int flags, ...)
     if (fd != -1) {
         if (fcntl(fd, F_SETFD, FD_CLOEXEC) == -1)
             av_log(NULL, AV_LOG_DEBUG, "Failed to set close on exec\n");
+        av_log(NULL, AV_LOG_INFO, "Disabling file cache\n");
+        fcntl(fd, F_NOCACHE, 1);
     }
 #endif
 
-- 
2.30.1 (Apple Git-130)

